platform/ethereum/ens.go

```diff
@@ -0,0 +1,47 @@
+package ethereum
+
+import (
+	"encoding/hex"
+	"net/http"
+
+	"github.com/spf13/viper"
+	"github.com/trustwallet/blockatlas/pkg/blockatlas"
+	"github.com/trustwallet/blockatlas/pkg/errors"
+
+	"github.com/ethereum/go-ethereum/ethclient"
+	ens "github.com/wealdtech/go-ens/v3"
+)
+
+func (p *Platform) Lookup(coin uint64, name string) (*blockatlas.Resolved, error) {
+	client, err := ethclient.Dial(viper.GetString("ethereum.rpc"))
+	result := blockatlas.Resolved{
+		Coin: coin,
+	}
+	if err != nil {
+		return nil, errors.E(http.StatusInternalServerError, "can't dial to ethereum rpc")
+	}
+	defer client.Close()
+
+	ensName, err := ens.NewName(client, name)
+	if err != nil {
+		return nil, errors.E(http.StatusInternalServerError, err.Error())
+	}
+
+	address, err := ensName.Address(coin)
+	if err != nil {
+		// return empty result for errors like: no unregistered
+		return &result, nil
+	}
+
+	result.Result = mapAddress(coin, address)
+	return &result, nil
+}
+
+func mapAddress(coin uint64, bytes []byte) string {
+	// FIXME: convert bytes to string according to coin
```

Comment:
Do we already have an issue with that? Use TODO, because the bot automatically create an issue for us

Reply:
no issue yet, I was planning to merge this first and create another one, looks like I can continue working on this branch

Reply:
done